ext/re2/extconf.rb

```diff
@@ -128,7 +128,7 @@ def build_extension
   have_header("stdint.h")
   have_func("rb_str_sublen")
 
-  unless have_library("re2")
+  if !static_p and !have_library("re2")
```

Comment:
Are we basically using our own variant of `have_library` when using MiniPortile2 here because we need to override the linker config to give our vendored dependencies priority?

Reply:
Yes, we're linking the `.a` explicitly instead of using `-lre2`. The latter will pick up the system dependency if it's installed in a certain location, and I don't believe there's anything we can do about that.